packages/swapper/src/api.ts

```diff
@@ -33,19 +31,96 @@ export type SupportedSellAssetsInput = {
   assetIds: CAIP19[]
 }
 
+export type CommonTradeInput = {
+  sellAsset: Asset
+  buyAsset: Asset
+  sellAmount?: string
+  buyAmount?: string
+  sendMax: boolean
+  sellAssetAccountId: string
+}
+export type GetTradeQuoteInput = CommonTradeInput
+
+export type BuildTradeInput = CommonTradeInput & {
+  buyAssetAccountId: string
+  slippage?: string
+  wallet: HDWallet
+}
+
+interface TradeBase<C extends ChainTypes> {
+  success: boolean // This will go away when we correctly handle errors
+  statusReason: string // This will go away when we correctly handle errors
+  buyAmount: string
+  sellAmount: string
+  feeData: chainAdapters.QuoteFeeData<C>
+  rate: string
+  allowanceContract: string
+  sources: Array<SwapSource>
+  buyAsset: Asset
+  sellAsset: Asset
+  sellAssetAccountId: string
+}
+
+export interface TradeQuote<C extends ChainTypes> extends TradeBase<C> {
+  minimum: string | null
+  maximum: string | null
+}
+
+export interface Trade<C extends ChainTypes> extends TradeBase<C> {
+  txData: string
+  depositAddress: string
+  receiveAddress: string
+}
+
+export type ExecuteTradeInput<C extends ChainTypes> = {
+  trade: Trade<C>
+  wallet: HDWallet
+}
+
+export type TradeResult = {
+  txid: string
+}
+
+export type SwapSource = {
+  name: string
+  proportion: string
+}
+
+export type ApproveInfiniteInput<C extends ChainTypes> = {
+  quote: Quote<C> | TradeQuote<C>
+  wallet: HDWallet
+}
+
+export type ApprovalNeededInput<C extends ChainTypes> = {
+  quote: Quote<C> | TradeQuote<C>
+  wallet: HDWallet
+}
+
 export class SwapError extends Error {}
 
 export interface Swapper {
   /** Returns the swapper type */
   getType(): SwapperType
 
+  /**
+   * Get builds a trade with definitive rate & txData that can be executed with executeTrade
+   **/
+  buildTrade(args: BuildTradeInput): Promise<Trade<ChainTypes>>
+
+  /**
+   * Get a trade quote
+   */
+  getTradeQuote(input: GetTradeQuoteInput, wallet?: HDWallet): Promise<TradeQuote<ChainTypes>>
+
   /**
    * Get a Quote along with an unsigned transaction that can be signed and broadcast to execute the swap
+   * @deprecated The method is going away soon.
    **/
   buildQuoteTx(args: BuildQuoteTxInput): Promise<Quote<ChainTypes>>
 
   /**
    * Get a basic quote (rate) for a trading pair
+   * @deprecated The method is going away soon.
    */
   getQuote(input: GetQuoteInput, wallet?: HDWallet): Promise<Quote<ChainTypes>>
```

Comment:
```suggestion
  getQuote(input: GetQuoteInput, wallet: HDWallet): Promise<Quote<ChainTypes>>
```

seems like wallet should a) always be required or b) not required at all?